### ISSUE TITLE
uf2 flash partially if not start at 4K boundary

### ISSUE THREAD

**Issue by juju2013** [raspberrypi/pico-bootrom#9]
Trying to do this:
```bash
dd if=/dev/urandom of=random.bin bs=1024 count=100
uf2conv.py -b 0x10020100 -o random.uf2 -c -f RP2040 random.bin
cp random.uf2 /mnt/uf2 && sync
```

Note that address starts at 0x10020100: after reboot, content between `0x****000` and `0x***100` stays unchanged (0xFFFFFFFF), shown as following:
( see https://gist.github.com/juju2013/6e5d129effd3e7d3b43a6bc8973306ff )
`
$10020100  ........ $10021000  FFFFFFFF $10021100  ........ $10022000  FFFFFFFF $10022100  ........ $10023000  FFFFFFFF $10023100  ........ $10024000  FFFFFFFF $10024100  ........ $10025000  FFFFFFFF $10025100  ........ $10026000  FFFFFFFF $10026100  ........ $10027000  FFFFFFFF $10027100  ........ $10028000  FFFFFFFF $10028100  ........ $10029000  FFFFFFFF $10029100  ........ $1002A000  FFFFFFFF
`
If that address starts at a 4K boundary, content flashed as expected.

Documentation says ( 2.8.3.2. UF2 Format Details ):
`
Note that flash is erased a 4K sector at a time, so writing to only a subset of a 4K flash sector will leave the rest of that flash sector undefined. Beyond that there is no requirement that a binary be contiguous.
`

**Comment by kilograham** [comment]
Please reformat that output!

**Comment by kilograham** [comment]
Yes this is a bug. UF2s that do not start on a 4K boundary are not written correctly.
elf2uf2 and the docs will be updated in the next release
